core/codegen/src/derive/form_field.rs

```diff
@@ -291,5 +292,39 @@ pub fn validators<'v>(
             })).unwrap()
         });
 
-        Ok(exprs)
+    Ok(exprs)
+}
+
+pub fn defaults<'v>(
+    field: Field<'v>,
+) -> Result<syn::Expr> {
```

Comment:
I think it would be nicer/cleaner if this returned a `Result<Option<syn::Expr>>` which is `Err` when the syntax is wrong, `Ok(Some)` when there's a default, and `Ok(None)` when there is no default. Then we can do the `or_else()` at compile-time, using the Rocket-default when there is no default specified.

This should also let you get away from the `flat_map().next()` complications, closer to how `one_field_name` is written.

Reply:
If this returns an `Option`, then I need to check whether a default was provided in `from_form.rs`, and emit different code when the default is there. I'll change it around a bit and we can see which approach we like better :+1: